lsp-mode.el

```diff
@@ -7081,22 +7126,21 @@ When prefix UPDATE? is t force installation even if the server is present."
 
 (defun lsp-async-start-process (callback error-callback &rest command)
   "Start async process COMMAND with CALLBACK and ERROR-CALLBACK."
-  (make-process
-   :name (cl-first command)
-   :command command
-   :sentinel (lambda (proc _)
-               (when (eq 'exit (process-status proc))
-                 (if (zerop (process-exit-status proc))
-                     (condition-case err
-                         (funcall callback)
-                       (error
-                        (funcall error-callback (error-message-string err))))
-                   (display-buffer " *lsp-install*")
-                   (funcall error-callback
-                            (format "Async process '%s' failed with exit code %d"
-                                    (process-name proc) (process-exit-status proc))))))
-   :buffer " *lsp-install*"
-   :noquery t))
+  (let ((name (cl-first command)))
+    (with-current-buffer (compilation-start (mapconcat #'shell-quote-argument command " ") t
+                                            (lambda (&rest _)
+                                              (generate-new-buffer-name (format "*lsp-install: %s*" name))))
+      (lsp-installation-buffer-mode +1)
+      (add-hook
```

Comment:
Don't you need to clean this lambda from `compilation-finish-functions`? Else the unrelated callback might be called for unrelated async command

Reply:
Since that hook is run only once, I'd say no. Are you worried about unecessary garbage?

Reply:
It seems that `compilation-finish-functions` is global variable, so if you run `lsp-async-start-process` multiple times to install multiple servers, shouldn't each call will add new closure to the hook?
Also, without checking the `_buf`, how can you make sure that the status is intended for the callback to react to?

Reply:
The fourth parameter, if set (as in this case) makes the hook buffer-local before setting it, and adds `t` to the hook variable given (making `run-hooks` run the global one as well).

Reply:
You're right, I didn't notice the `local` flag being set